lib/matplotlib/tests/test_ticker.py

```diff
@@ -452,23 +452,111 @@ def test_percentformatter():
 
 def test_EngFormatter_formatting():
     """
-    Create two instances of EngFormatter with default parameters, with and
-    without a unit string ('s' for seconds). Test the formatting in some cases,
-    especially the case when no SI prefix is present, for values in [1, 1000).
+    Test instances of EngFormatter for different values of the kwarg `places`,
+    with and without a unit symbol, as well as with and without a space
+    separator between the value and the prefix/unit.
 
-    Should not raise exceptions.
+    Should not exception.
     """
-    unitless = mticker.EngFormatter()
-    nose.tools.assert_equal(unitless(0.1), u'100 m')
-    nose.tools.assert_equal(unitless(1), u'1')
-    nose.tools.assert_equal(unitless(999.9), u'999.9')
-    nose.tools.assert_equal(unitless(1001), u'1.001 k')
-
-    with_unit = mticker.EngFormatter(unit=u's')
-    nose.tools.assert_equal(with_unit(0.1), u'100 ms')
-    nose.tools.assert_equal(with_unit(1), u'1 s')
-    nose.tools.assert_equal(with_unit(999.9), u'999.9 s')
-    nose.tools.assert_equal(with_unit(1001), u'1.001 ks')
+    UNIT = u's'
+    DIGITS = set([u'{0:d}'.format(d) for d in range(10)])
+
+    # Case A/ With kwarg `places=None` (default)
+    A_wo_unit_w_sep = ((-1234.56789, u'-1.23457 k'),
```

Comment:
To be honest, the expected string in the cases `places=0` and `places=2` correspond to what was returned before this PR, but I find that some of the returned strings are weird. Like for example with `places=0`, 999.9 -> u'999' and 987654.321 -> u'987 k': shouldn't it be u'1000' , resp. u'988' instead? 

Besides, I'm not sure the trick with `DIGIT` is the cleanest one (or at least the more readable one).